src/util/strencodings.cpp

```diff
@@ -81,16 +81,21 @@ template <typename Byte>
 std::optional<std::vector<Byte>> TryParseHex(std::string_view str)
 {
     std::vector<Byte> vch;
-    auto it = str.begin();
-    while (it != str.end()) {
-        if (IsSpace(*it)) {
-            ++it;
-            continue;
-        }
-        auto c1 = HexDigit(*(it++));
-        if (it == str.end()) return std::nullopt;
-        auto c2 = HexDigit(*(it++));
-        if (c1 < 0 || c2 < 0) return std::nullopt;
+    vch.reserve(str.size() / 2);
+
+    for (auto it = str.begin(); it != str.end(); ++it) {
+        const auto c1{HexDigit(*it)};
+        if (c1 < 0) {
+            if (IsSpace(*it)) {
+                continue;
+            } else {
+                return std::nullopt;
+            }
+        } else if (++it == str.end()) return std::nullopt;
```

Comment:
The style is a bit all over the place (I tried to adhere to the previous style), should I add braces everywhere in this method?

Reply:
It took me a minute to figure out where `it` was being incremented when you put it in the `else if`. Not a big deal but maybe add a break here and move this check above where we define `c2` so it's more clear that is where we are moving `it` forward?

```suggestion
        }
        
        if (++it == str.end()) return std::nullopt;
```

Reply:
I don't think you need braces for a return early though.

Reply:
Rebased the last commit to keep the history clean. 
hanks for taking the time to review this PR thoroughly, appreciate it.